### ISSUE TITLE
Overlay position that's attached to an input is incorrect

### ISSUE THREAD

**Issue by frontsideair** [react-bootstrap/react-overlays#57]
I have a Popover which is wrapped by an Overlay element and it has a text input element as the target, and that Popover is placed top left-ish corner of the page, which is not the correct place. I think the problem is not within `react-bootstrap`, so I opened the issue here.

I checked the arguments provided to `calcOverlayPosition` code for possible bugs but it looks like the `target` is a direct child of the `container`, which seems okay to me. Any idea how to solve it?

**Comment by taion** [comment]
Double-check your code; this use case is supported just fine – most likely some of your positioning props are getting lost somewhere.

**Comment by frontsideair** [comment]
Here's the code: https://github.com/frontsideair/react-clockpicker/blob/master/src/ClockPicker.js#L131

`container` is `this`, which refers to the `div`, `Input` is the `target`, which is the direct descendant of `container`, and `Overlay` is a sibling of the `target`. I think this must be the correct configuration, but it's still misplaced.

**Comment by frontsideair** [comment]
I finally solved the problem.

I used OverlayTrigger (which had correct placement) and observed differences between the way I was using Overlay. The only difference was the container I was providing was `this`, where OverlayTrigger was providing `body`. :open_mouth: I did the same and it worked.

Is it the expected behavior? Should the docs be updated with this information?

By the way, the problem was not limited to `input`, it was incorrect for a `Button` as well.
